src/print.rs

```diff
@@ -0,0 +1,318 @@
+use crate::config::{LanguageSettings, ParserSettings};
+use crate::document::code::{CodeBlock, Line, Source};
+use crate::document::transclusion::Transclusion;
+use crate::document::{CompileError, CompileErrorKind, Document, Node};
+use crate::parser::code::RevCodeBlock;
+use std::collections::HashMap;
+use std::fmt::Write;
+
+/// Formats this `Document` as a string containing the documentation file contents
+pub fn print_docs(document: &Document, settings: &ParserSettings) -> String {
+    let mut output = String::new();
+    for node in &document.nodes {
+        match node {
+            Node::Transclusion(transclusion) => {
+                output.push_str(&print_transclusion(transclusion, false))
```

Comment:
To avoid all these intermediate allocations, it would probably be best if all `print_*` functions would take a `fmt: &mut fmt::Formatter` argument instead of producing a `String`. This would ensure that they combine naturally but producing a `String` as the result would still be simple.

Reply:
What should I then use as argument? To use `String`, it would require `dyn Write` instead of `Formatter`...

Reply:
Either generics like `&mut impl Write` or as just suggest `&mut dyn Write`. Since we will probably use each function in once specific case, using `&mut impl Write` seems preferable as there is little fear of code bloat and hence no reason to eat the overhead of dynamic dispatch. (Note that `&mut impl Write` is basically shorthand for `&mut F` and `where F: Write`.)

Reply:
> (Note that `&mut impl Write` is basically shorthand for `&mut F` and `where F: Write`.)

I was already starting to ask myself why not do it this way 😆